test/functional/test_runner.py

```diff
@@ -157,6 +157,32 @@
 # Place EXTENDED_SCRIPTS first since it has the 3 longest running tests
 ALL_SCRIPTS = EXTENDED_SCRIPTS + BASE_SCRIPTS
 
+def check_script_prefixes():
+    """Check that no more than `EXPECTED_VIOLATION_COUNT` of the
+       test scripts don't start with one of the allowed name prefixes."""
+    EXPECTED_VIOLATION_COUNT = 75
+    LEEWAY = 10
+
+    ok = {"feature", "interface", "mempool", "mining",
+          "p2p", "rpc", "wallet"}
+    excess = set(x.split(" ",1)[0] for x in ALL_SCRIPTS)
+    excess.discard("example_test.py")
+
+    for prefix in ok:
+        excess -= set(x for x in excess if x.startswith(prefix+"_"))
+
+    if len(excess) < EXPECTED_VIOLATION_COUNT:
+        print("{}HURRAY!{} Number of functional tests violating naming convention reduced!".format(BOLD[1], BOLD[0]))
+        print("Consider reducing EXPECTED_VIOLATION_COUNT from %d to %d" % (EXPECTED_VIOLATION_COUNT, len(excess)))
+
+    assert len(excess) <= EXPECTED_VIOLATION_COUNT + LEEWAY, "Too many tests not following naming convention! (%d found, expected: <= %d)" % (len(excess), EXPECTED_VIOLATION_COUNT)
+
+    if 0 < len(excess) <= LEEWAY:
```

Comment:
bug: this should be `EXPECTED_VIOLATION_COUNT < len(excess) <= EXPECTED_VIOLATION_COUNT + LEEWAY`